src/core/init/manifest_update_scheduler.ts

```diff
@@ -204,9 +194,70 @@ export default function manifestUpdateScheduler({
         }
         return handleManifestRefresh$(evt);
       }));
-  }
 
-  return observableDefer(() => handleManifestRefresh$(initialManifest));
+    /**
+     * Create an Observable that will emit when the Manifest needs to be
+     * refreshed according to the Manifest's internal properties (parsing
+     * time is also taken into account in this operation to avoid refreshing too
+     * often).
+     * @returns {Observable}
+     */
+    function createAutoRefreshObservable() : Observable<{
+      completeRefresh: boolean;
+      unsafeMode: boolean;
+    }> {
+      if (manifest.lifetime === undefined || manifest.lifetime < 0) {
+        return EMPTY;
+      }
+
+      /** Regular refresh delay as asked by the Manifest. */
+      const regularRefreshDelay = manifest.lifetime * 1000 - timeSinceRequest;
+
+      /** Actually choosen delay to refresh the Manifest. */
+      let actualRefreshInterval : number;
+
+      if (totalUpdateTime === undefined) {
+        actualRefreshInterval = regularRefreshDelay;
+      } else if (manifest.lifetime < 3 && totalUpdateTime >= 100) {
+        // If Manifest update is very frequent and we take time to update it,
+        // postpone it according to the maximum value between:
+        actualRefreshInterval = Math.max(
+          // Take 3 seconds as a default safe value for a base interval.
+          3000 - timeSinceRequest,
+
+          // Add update time to the original interval.
+          Math.max(regularRefreshDelay, 0) + totalUpdateTime,
+
+          // Limit the postponment to a very high value relative to
+          // `regularRefreshDelay`.
+          // This avoid perpetually postponing a Manifest update when
+          // performance seems to have been abysmal one time.
+          regularRefreshDelay * 6);
+        log.info("MUS: Manifest update rythm is too frequent. Postponing next request.",
+                 regularRefreshDelay,
+                 actualRefreshInterval);
+      } else if (totalUpdateTime >= (manifest.lifetime * 1000) / 10) {
+        // If Manifest updating time is very long relative to its lifetime,
+        // postpone it:
+        actualRefreshInterval = Math.max(
+          // Just add the update time to the original waiting time
+          Math.max(regularRefreshDelay, 0) + totalUpdateTime,
+
+          // Limit the postponment to a very high value relative to
+          // `regularRefreshDelay`.
+          // This avoid perpetually postponing a Manifest update when
+          // performance seems to have been abysmal one time.
+          regularRefreshDelay * 6);
```

Comment:
Maybe we should check that regularRefreshDelay * 6 is higher than the total update time :

js```
actualRefreshInterval = Math.min(
  Math.max(regularRefreshDelay, 0) + totalUpdateTime,
  totalUpdateTime,
  regularRefreshDelay * 6)
)
```

Reply:
If I did that, this PR would not have much effect I think.
I wanted to limit in any case the refresh rate to 6 times the regular update time (not considering emsg, minimumManifestUpdateInterval etc.)